init.rb

```diff
@@ -10,7 +10,7 @@ module ValidatesUrlFormatOf
     \A
     https?://                                                          # http:// or https://
     ([^\s:@]+:[^\s:@]*@)?                                              # optional username:pw@
-    ( ((#{ALNUM}+\.)*xn---*)?#{ALNUM}+([-.]#{ALNUM}+)*\.[a-z]{2,6}\.? |  # domain (including Punycode/IDN)...
+    ( ((#{ALNUM}+\.)*xn---*)?#{ALNUM}+([-.]#{ALNUM}+)*\.[a-z]{2,6}\.?(--#{ALNUM}*)? |  # domain (including Punycode/IDN)...
```

Comment:
Thank you for this PR, and sorry for not getting to it sooner.

On a minor note, does the below proposed change make sense? I assume we want one-or-more rather than zero-or-more.

```suggestion
    ( ((#{ALNUM}+\.)*xn---*)?#{ALNUM}+([-.]#{ALNUM}+)*\.[a-z]{2,6}\.?(--#{ALNUM}+)? |  # domain (including Punycode/IDN)...
```